packages/react-components/react-tree/src/components/Tree/useTreeContextValues.ts

```diff
@@ -1,22 +1,41 @@
+import * as React from 'react';
 import { TreeContextValue } from '../../contexts';
 import { TreeContextValues, TreeState } from './Tree.types';
 
 export function useTreeContextValues_unstable(state: TreeState): TreeContextValues {
-  const { openItems, checkedItems, selectionMode, level, appearance, size, requestTreeResponse, treeType } = state;
-  /**
-   * This context is created with "@fluentui/react-context-selector",
-   * there is no sense to memoize it
-   */
-  const tree: TreeContextValue = {
-    treeType,
-    size,
-    level,
-    openItems,
-    appearance,
-    checkedItems,
-    selectionMode,
-    requestTreeResponse,
-  };
+  if (state.contextType === 'root') {
+    const {
+      openItems,
+      level,
+      contextType: type,
+      treeType,
+      checkedItems,
+      selectionMode,
+      appearance,
+      size,
+      requestTreeResponse,
+    } = state;
+    /**
+     * This context is created with "@fluentui/react-context-selector",
+     * there is no sense to memoize it
+     */
+    const tree: TreeContextValue = {
+      treeType,
+      size,
+      openItems,
+      appearance,
+      checkedItems,
+      selectionMode,
+      contextType: type,
+      level,
+      requestTreeResponse,
+    };
 
-  return { tree };
+    return { tree };
+  }
+  return {
+    // level is statically determined by the context
```

Comment:
the comment is not useful there as the condition is based on `state.contextType`